lib/omnibus/software.rb

```diff
@@ -581,6 +582,14 @@ def builder
       @builder ||= Builder.new(self)
     end
 
+    def manifest_entry
+      Omnibus::ManifestEntry.new(name, {
+                                   source_type: source_type,
+                                   described_version: version,
+                                   locked_version: fetcher.resolved_version,
+                                   locked_source: source})
```

Comment:
A couple of thoughts here:

* Since we are converting to a manifest entry maybe a method name of `to_manifest_entry` makes sense.
* Should we cache this?
* Setting attributes/properties via a hash in the initializer could prove error prone. Why don't we just set values using the accessor?

```ruby
def manifest_entry
  @manifest_entry ||= begin
    manifest_entry = Omnibus::ManifestEntry.new(name)
    manifest_entry.source_type = source_type
    manifest_entry.described_version = version
    manifest_entry.locked_version = fetcher.resolved_version
    manifest_entry.locked_source = source
    manifest_entry
  end
end
```